src/sentry/shared_integrations/client/base.py

```diff
@@ -329,3 +333,64 @@ def get_with_pagination(
             if num_results < page_size:
                 return output
         return output
+
+    def record_response(self, response: Response | None = None, error: Exception | None = None):
+        if not hasattr(self, "integration_id"):
+            return
+        if self.is_response_error(response, error):
+            self.record_request_error(response, error)
+        elif self.is_response_success(response):
+            self.record_request_success(response)
+        elif self.is_response_fatal(response, error):
+            self.record_request_fatal(response, error)
+
+    def record_request_error(self, resp: Response | None = None, error: Exception | None = None):
+        if not hasattr(self, "integration_id"):
+            return
+        if not self.integration_id:
+            return
+        if error is None:
+            error = resp.get("error", None)
+        if not self.is_response_error(resp, error):
+            return
+        buffer = IntegrationRequestBuffer(self._get_redis_key())
+        buffer.record_error()
+        print("error recorded")
+        if buffer.is_integration_broken():
+            self.uninstall_integration(self.integration_id)
+
+    def record_request_success(self, resp: Response):
+        if not hasattr(self, "integration_id"):
+            return
+        if not self.is_response_success(resp):
+            return
+        buffer = IntegrationRequestBuffer(self._get_redis_key())
+        buffer.record_success()
+        print("success recorded")
+
+    def record_request_fatal(self, resp: Response | None = None, error: Exception | None = None):
+        if not hasattr(self, "integration_id"):
+            return
+        if not self.is_response_fatal(resp, error):
+            return
+        buffer = IntegrationRequestBuffer(self._get_redis_key())
+        buffer.record_fatal()
+        print("fatal recorded")
+        if buffer.is_integration_broken():
+            self.disable_integration(self.integration_id)
+
+    def disable_integration(self, integration_id: int) -> None:
+        rpc_integration, rpc_org_integration = integration_service.get_organization_context(
+            integration_id=self.integration_id
```

Comment:
I'm a little confused here - this function is being passed `integration_id` but you're using `self.integration_id`. Does it need to be passed or can we get it from `self`? I don't think we need both.